localparticipant.go

```diff
@@ -359,16 +361,7 @@ func (p *LocalParticipant) updateInfo(info *livekit.ParticipantInfo) {
 			continue
 		}
 		if pub.IsMuted() != ti.Muted {
-			pub.SetMuted(ti.Muted)
-
-			// trigger callback
-			if ti.Muted {
-				p.Callback.OnTrackMuted(pub, p)
-				p.roomCallback.OnTrackMuted(pub, p)
-			} else if !ti.Muted {
-				p.Callback.OnTrackUnmuted(pub, p)
-				p.roomCallback.OnTrackUnmuted(pub, p)
-			}
+			_ = p.engine.client.SendMuteTrack(pub.SID(), pub.IsMuted())
```

Comment:
Should this be called directly?  Feels like `pub.SetMuted` would be the right thing as it will do the right thing of sending message and issuing callback? Did you want to remove the callback when mute is managed through this path? Even then, calling `SetMuted` and doing callback or not there might be better?

Reply:
The pub.SetMuted will find the new state is the same as the current publication state and return directly, this pr keeps same behavior like js-client to send a mute message only when server-side info is inconsistent